### DISCUSSION TITLE
primary key id does not increment on insert and doesn't start from last row's id

### DISCUSSION THREAD

**Discussion by johnatr** [sqlalchemy/sqlalchemy#7954] [Usage Questions] [answered]
Using SQLModel and SQLAlchemy, I have an existing table with rows of data that I am trying to model into it's own class. I've google around, read SQLModel docs but cannot find out how to automatically increment each insert's primary key from the last available row, like I'd typically expect when writing a raw `INSERT` sql query.

    from datetime import datetime, date, time
    from sqlmodel import Field, Integer, SQLModel, null
    from sqlalchemy import String, inspect
    from sqlalchemy.sql.schema import Column
    
    class Movies(SQLModel, table=True):
        movieId: int = Field(sa_column=Column('movieId', Integer, autoincrement='auto', 
                                                primary_key=True, default=0, unique=True))
        title: str
        rating: int
    
        engine = create_engine(mysql_url, echo=True)
    
    
            with Session(engine) as session:
    
                movieInsert = Movie(movieId='', title='Gone', rating=5)
                session.add(movieInsert)
                
                session.commit()

Instead of incrementing the `movieId` from the last row 382, it is inserting 0 each time.

How do I make it insert 383, and on the next insert 384, and so on? If this is not possible, then how can I specify a number for it to start from?

Do I need to grab the last row, get its id, and then add? Kind of surprised how much more work this is compared to just writing a simple INSERT query that MySQL automatically takes. care of but I'm sure I'm missing some key sql alchemy info

**Comment by CaselIT** [comment, marked as answer]
Hi,

Try removing `default=0` since that indicates to sqlalchemy that if no value is provided by the user that value should be used.

**Reply by johnatr** [reply]
Thanks for answering. However if I don't pass that default=0 it will complain

sqlalchemy.exc.IntegrityError: (MySQLdb._exceptions.IntegrityError) (1364, "Field 'movieId' doesn't have a default value")

**Reply by CaselIT** [reply]
From the error it seems that movieId is not defined as a primary key that can increment itself.
How was the table generated?
